packages for managing clinical trial data (#273)
pharmaverse/blog: posts/2025-02-28_theres_a_pharmaverse_package_for_that/managing-clinical-trial-data.qmd

## Key Stages of Clinical Reporting

Managing clinical trial data involves multiple stages, each with its own challenges. **Pharmaverse** provides a range of R packages that support different parts of the process, sometimes even offering multiple options for the same task. This flexibility allows organizations to choose the best tools for their specific needs rather than sticking to a one-size-fits-all approach.

A **metadata-driven approach** helps ensure that clinical trial data is consistently structured and aligned with regulatory standards. The typical process follows this sequence:

**Metadata** ➝ **OAK** ➝ **Admiral** ➝ **Define.xml** ➝ **TLGs** ➝ **Submissions**

Some examples of **pharmaverse** packages that support clinical reporting include:

-   {diffdf} – Tracking differences in datasets.
-   {metatools} – Metadata management and transformation.
-   {sdtm.oak} – The primary **pharmaverse** package for SDTM dataset creation.
-   {datcur} - Performind datacuts.
-   {admiral} – Standardized data derivations.
-   {metacore} – Metadata-driven structures.

> -   The **pharmaverse** provides multiple table-making packages, such as {chevron} (which builds on {rtables}), {tplyr}, {pharmaRTF}, {gtsummary}, {cards}, {tfrmt}, and {tidytlg}. More tools are listed on the [TLGs page](https://pharmaverse.org/e2eclinical/tlg/).

Reviewer: ```suggestion
-   The **pharmaverse** provides multiple table-making packages, such as {chevron} (which builds on {rtables}), {Tplyr}, {pharmaRTF}, {gtsummary}, {cards}, {tfrmt}, and {tidytlg}. More tools are listed on the [TLGs page](https://pharmaverse.org/e2eclinical/tlg/).
```

@mstackhouse would shout at me if i missed that one! ;P

    by_vars = exprs(STUDYID, USUBJID),
    new_var = SAFFL,
    condition = EXDOSE > 0 | str_detect(EXTRT, "PLACEBO")
  )
```

#### **Step 5: Generate and Save Results**

Finally, we save the dataset CSV and can view some of its columns:

```{R, eval = FALSE}
# Save to a CSV file
write.csv(adsl, "adsl_output.csv", row.names = FALSE)

adsl
```

```{r, eval=TRUE, echo=FALSE}
adsl%>%
  dataset_vignette(
    display_vars = exprs(USUBJID, TRT01P, TRT01A, TRTSDT, TRTEDT, SAFFL)
  )
```

#### **More Details on ADSL Creation**

This is just a **high-level example**; the full process includes deriving death variables, grouping populations, and applying labels. For a deeper dive, check out the [ADSL Implementation Guide](https://cran.r-project.org/web/packages/admiral/vignettes/adsl.html).

## **Who Are the Key Players in Pharmaverse, and Do You Need to Use All Packages?**

### **Key Players in pharmaverse**

-   **Pharmaverse Council and Community** – A collaborative group of developers, industry experts, and contributors maintaining and expanding the ecosystem.
-   **Open-Source Contributors** – Individuals and organizations developing and refining **pharmaverse** packages.

> -   **Pharmaverse is supported by [PHUSE](https://phuse.global/)** – PHUSE plays an active role in supporting and advancing the **pharmaverse** initiative.

Reviewer: ```suggestion
-   **Pharmaverse is part of [PHUSE](https://phuse.global/)** – PHUSE plays an active role in supporting and advancing the **pharmaverse** initiative.
```
